### ISSUE TITLE
Update/remove all with auto-completion

### ISSUE THREAD

**Issue by soncodi** [typeorm/typeorm#999]
Is there a simple syntax to update or remove multiple entities via Active Record or Data-Mapper? I'm looking for something like (in sequelize):

`await User.update({ name: 'a' }, { where: { name: 'b' } });`
`// UPDATE users SET name = 'a' WHERE name = 'b';`

and

`await User.update({ name: 'a' });`
`// UPDATE users SET name = 'a';`

EntityManager throws `Cannot find entity to update by a given criteria` if there are no matches, and only updates one record:

`await connection.manager.update(User, { name: 'a' }, { name: 'b' });`
`await connection.manager.update(User, { where: { name: 'a'  } }, { name: 'b' });`

QueryBuilder works, but is verbose and has no type checking on `.where()`:

```
await User.createQueryBuilder('user')
      .update({ name: 'b' })
      .where('user.name = :name', { name: 'a' }) // hard to maintain, no type check
      .execute();
```

**Comment by daniel-lang** [comment]
Repositories have an update function that has a very similar syntax. Only, that the parameters are switched (first the condition and than the part you want to change)

```typescript
await repository.update({ firstName: "Timber" }, { firstName: "Rizzrak" });
// executes UPDATE user SET firstName = Rizzrak WHERE firstName = Timber
```

**Comment by pleerock** [comment]
@daniel-lang is right, in your case its  a method from `BaseEntity`:

```ts
await User.update({ firstName: "Timber" }, { firstName: "Rizzrak" });
```

**Comment by soncodi** [comment]
I tried the following:

`await User.update({ name: 'a' }, { name: 'b' });`

Logger results:

    executing query:  SELECT `User`.`id` AS `User_id`, `User`.`name` AS `User_name`, `User`.`email` AS `User_email`, `User`.`password` AS `User_password` FROM `users` `User` WHERE `User`.`name`=? -- PARAMETERS: ["a"]
    executing query:  START TRANSACTION
    executing query:  UPDATE `users` SET `name`=?  WHERE `id`=? -- PARAMETERS: ["b",2]
    executing query:  COMMIT

Observed this behavior:
- issues a SELECT first
- only updates the first user.
- errors once no users matched `Error: Cannot find entity to update by a given criteria`

**Comment by daniel-lang** [comment]
You are right. Looking at the source code it executes `findOne()` and than updates the found entity if it exists, otherwise throws an error.
So even though it's the same syntax, the `update` function does something every different. Sorry, for suggesting it, I should have checked the source code.

So I guess the only way to do what you want is with the QueryBuilder.
@pleerock is that true?

**Comment by soncodi** [comment]
It seems the goal is to act as an entity cache layer in front of the db. That's valid design but, for my use-case, it would be great to be able to opt-out of that behavior, and just use TypeORM as a simple mapper with type-checking. It could essentially be a drop-in replacement for Sequelize. I assumed this was implied by disabling `cache` on the connection, but I have not yet looked through the code.

**Comment by pleerock** [comment]
`update` method currently using `find` and `save methods. Instead it should just use update query builder functionality. Ive got sick, thats why release is delayed this week, I'll fix update methods next week and publish a new version with those changes.

**Comment by soncodi** [comment]
@pleerock perfect, thanks!

**Comment by pleerock** [comment]
I have updated `update` methods. Now they should work fine. This change will be released in `0.1.1`
